email: Make HTML versions of emails that don't have it.

RameshAditya opened the pull request:
Added html versions of emails missing one, having only txt files. Updated the test_templates.py file to test the new html files as well.

Fixes #4691

smarx commented:
*Automated message from Dropbox CLA bot*

@RameshAditya, it looks like you've already signed the Dropbox CLA. Thanks!

rht commented on templates/zerver/emails/confirm_registration_mit.html:
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
```
I have seen that this is the case in the existing templates, but shouldn't it be better with `<!DOCTYPE html>`?

rht replied:
Found a cleaner reference point here: https://github.com/zulip/zulip/blob/44113d6436da02ef4126aab8342befe53605fe8d/templates/zerver/base.html. For instance, the next line SHOULD be `<html lang='{{LANGUAGE_CODE}}'>`.

rht replied:
@RameshAditya  sorry, disregard these comments, since the fix is outside of the scope of this PR. I will fix them later or if you want to tackle them as well.

rht commented on templates/zerver/emails/confirm_registration_mit.html:
```diff
@@ -0,0 +1,37 @@
+                        There are a few steps you need to take to ensure Zulip works with your MIT Zephyr account.
+                    </p>
+                    <p>
+                        1) Create a password and activate your account:<br />
```
This should be an ordered list, wrapped with `<ol>`.

rht commented on templates/zerver/emails/password_reset.html:
```diff
@@ -0,0 +1,27 @@
+                    <p>
+                        {% if attempted_realm %}
+                        Someone (possibly you) requested a password reset email for {{ email }} on {{ attempted_realm.uri }}, but {{ email }} does not have an active account in {{ attempted_realm.uri }}.  However, {{ email }} does have an active account in {{ user.realm.uri }} organization; you can try logging in or resetting your password there.
+                        {% else %} Psst. Word on the street is that you forgot your password, {{ email }}.<br />
```
Hmm? Why is the newline after `{% else %}` missing?

RameshAditya commented:
Apologies for the delay, I've added the newline after the `{% else %}` line in the password_reset.html and I've changed the confirm_registration_mit.html file's paragraph format to an ordered list.

rishig commented:
Thanks for your work on this @RameshAditya, and for your comments @rht! And also sorry for the long delay in getting this merged; we've been on the road (at PyCon) for the last two weeks.

I agree with cleaning up the header if needed; it should be easy to iterate on after https://github.com/zulip/zulip/issues/5114 is resolved.

I squashed your commits and put them at https://github.com/rishig/zulip/tree/review-4790; @timabbott, ready to merge. I didn't manually test (for newlines, etc), but given the large number of changes in the pipeline I imagine much of the text is about to be rewritten anyway.

timabbott commented:
@rishig I neglected to merge these when you posted that 1.5 weeks ago; do you think it makes sense to combine this with the #5114 at this point?

rishig commented:
yeah, I'll rebase this and repost.

rishig commented:
Rebased onto #5335. Closing.

timabbott commented:
This was just merged, thanks @RameshAditya for your work on this!!

RameshAditya commented:
I learnt a lot, doing this and I intend to try my hand on more bugs. The pleasure is mine. :)

timabbott commented:
@RameshAditya awesome!  And congrats on your first code contribution to Zulip :)
